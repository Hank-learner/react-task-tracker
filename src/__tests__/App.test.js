import { render, screen, fireEvent } from '@testing-library/react';
import App from '../App';

describe("Main page snapshot", () => {
    it('should take a snapshot', () => {
        const { asFragment } = render(<App />)
        expect(asFragment(<App />)).toMatchSnapshot()
    });
})

describe("rendering", () => {
    test('renders header', () => {
        const { getByText } = render(<App />);
        const headerElement = screen.getByText(/task tracker/i);
        expect(headerElement).toBeInTheDocument();
        expect(getByText('Task Tracker', { exact: false })).not.toBeNull();
    });

    test('shouldnt render addTask', () => {
        const { queryByLabelText } = render(<App />);
        expect(queryByLabelText('Enter Task')).toBeNull();
    });

    test('renders addTasks btn', () => {
        const { getByText, queryByLabelText } = render(<App />);
        fireEvent.click(getByText('Add'))
        expect(getByText('Close')).not.toBeNull();
        fireEvent.click(getByText('Close'))
        expect(queryByLabelText('Enter Task')).toBeNull();
    });
})


// test('renders', () => {
//     const root = document.createElement('div');

//     ReactDom.render(<App />, root)

//     expect(root.querySelector('header h1').textContent).toBe(
//         "Task Tracker"
//     );


//     const {getByText, getByLabelText} = getQueriesForElement(root); 

//     //check if heading present or not
//     expect(getByText('Task Tracker')).not.toBeNull();

//     //form check label matches the feild id (htmlFor)
//     // expect(getByLabelText('Enter Task')).not.toBeNull();

//     ReactDom.unmountComponentAtNode(root);
// });
