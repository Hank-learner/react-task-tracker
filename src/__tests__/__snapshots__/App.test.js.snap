// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <header
      class="header"
      id="header"
    >
      <h1>
        Task Tracker
      </h1>
      <button
        class="btn"
        data-testid="add-task"
        style="background-color: green;"
      >
        Add
      </button>
    </header>
    No Tasks to Show
    <footer>
      <p>
        Developed with help of Traversy Media Â© 2021
      </p>
      <a
        href="/about"
      >
        About
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
